src/libchdr_chd.c

```diff
@@ -1980,10 +2064,7 @@ CHD_EXPORT chd_error chd_precache(chd_file *chd)
 
 	if (chd->file_cache == NULL)
 	{
-		size = core_fsize(chd->file);
```

Comment:
The size variable is no longer initialized by this change, leading to undefined behaviour when it's accessed further down.

Reply:
Ugh, C89 strikes again. Indeed, this would break precaching. I'm using a rewritten version of this function in DS with progress reports back, hence why I never noticed...